### PR TITLE
v.2022-06-a

### PR DESCRIPTION
**Pull request by rcv-aws**
### Description
v.2022-06-a, various updates
 
### Issues Resolved
- consume XML files with queries captured by SQL Server Profiler.
- mark cursor variables as supported since v.1.0.0.
- better detection of variable dependencies in SELECT.
- detect table-valued functions using SELECT TOP.
- detect INSERT..EXECUTE() on a single string constant (which is supported).
- the -rewrite option now handles unquoted strings and double-quoted strings with embedded quotes.
- detect syntax issues with UPDATE/DELETE statements when specifying -syntax_issues option (experimental).
- performance improvements when analyzing large number of applications together.
- reporting improvements with better hyperlinks.
- various small fixes.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish_internal/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+# 2022-06-a
+- consume XML files with queries captured by SQL Server Profiler.
```
Comment (hsuamz) [inline CHANGELOG.md:2]: nit: A styling comment. The rest of the changelog starts each sentence capitalized. Suggest we follow the same format.

**Review by hsuamz — COMMENTED** [review]
Small nit with the changelog, everything else LGTM.

File: src/main/java/compass/Compass.java [hunk 2]
```diff
@@ -1559,39 +1619,83 @@ private void processInput(String runStartTime) throws Exception {
...
+			} 
+			else {
+				// already-imported files: need to sort on app name + original src file path
+				String app = u.getAppNameFromImported(f);						
+				String origSrcFile = u.importFileAttribute(u.importFileFirstLine(f), 1);						
+				sortInputFiles.add(app.toUpperCase() + sortKeySeparator + origSrcFile + sortKeySeparator + f);
+			}
+
+			if ((u.analysisPass == 1) && (!reAnalyze)) {
+				// remove duplicate input files during finitial import only
```
Comment (avneettoor) [inline src/main/java/compass/Compass.java:1662]: Nit: spelling. I think you meant initial?

File: src/main/java/compass/CompassUtilities.java [hunk 3]
```diff
@@ -3171,47 +3402,69 @@ public String getObjectNameFromID(String ID) {
...
+		if (!tmp.contains(".")) {
+			return tmp;
 		}
-		DBName = parts.get(parts.size()-3);
-		return DBName;
+		return tmp.substring(tmp.lastIndexOf(".")+1);		
 	}
 
 	// extract remote servername from 4-part object name
+	// this is called only when it is know that it's a 4-part name, so this call does nto happen so often
```
Comment (avneettoor) [inline src/main/java/compass/CompassUtilities.java:3458]: Nit: spelling of `not`

File: src/main/java/compass/Compass.java [hunk 4]
```diff
@@ -1268,8 +1325,9 @@ private static boolean optionsValid() throws Exception {
 		}
 		
 		if (reAnalyze && (!generateReport)) {
-			u.appOutput("Cannot combine -analyze and -noreport");
-			return false;
+			// actually, no reason to disallow this: runnign report generation separately can be significantly faster for very large cases
```
Comment (avneettoor) [inline src/main/java/compass/Compass.java:1328]: Nit: spelling of `running`

File: CHANGELOG.md (hunk 1 above)
Comment (rcv-aws) [reply CHANGELOG.md]: done

**Review by avneettoor — APPROVED** [review]

**Review by hsuamz — APPROVED** [review]

**Pull request merged**